legal-api/src/legal_api/services/document_meta.py

```diff
@@ -286,7 +286,10 @@ def get_alteration_reports(self, filing: dict):  # pylint: disable=no-self-use
                     DocumentMetaService.ReportType.ALTERATION_NOTICE.value
                 )
             )
-            if 'nameRequest' in filing['filing']['alteration']:
+            name_request = filing.get('filing', None).get('alteration', None).get('nameRequest', None)
+            business = filing.get('filing', None).get('business', None)
+            if name_request and 'legalName' in name_request and \
+                    name_request['legalName'] != business.get('legalName', None):
```

Comment:
Did you check the change of name to a numbered company? That should generate a certificate as well.

Reply:
Yes I did check the existing filing json. This should work